### ISSUE TITLE
Heka may generate too many logs

### ISSUE THREAD

**Issue by elemoine** [mozilla-services/heka#1894]
With full_action="drop" Heka will [write](https://github.com/mozilla-services/heka/blob/e99096e737959d6991ceed8a962c2aa1eefa1d93/pipeline/router.go#L340-L341) a "Queue is full" error to its log (hekad.log) **for each** dropped message. On our lab we've observed a rate of 4 GB of "Queue is full" error messages per hour, which is a real problem.  Heka should probably apply some throttling for this error message.

**Comment by elemoine** [comment]
This is the sort of local patch we're going to use for now: https://github.com/mozilla-services/heka/compare/v0.10.0...elemoine:ratelimit (based on the v0.10.0 tag).

**Comment by rafrombrc** [comment]
This is great, but I don't want to introduce another dependency, would rather just embed the rate limiting code into Heka directly (w/ credit and link to original source, of course).

**Comment by elemoine** [comment]
Yeah, I needed a quick fix. I'll see if I can come up with a proper patch.
